src/sdam/topology.ts

```diff
@@ -347,38 +347,46 @@ export class Topology extends EventEmitter {
     // connect all known servers, then attempt server selection to connect
     connectServers(this, Array.from(this.s.description.servers.values()));
 
-    const readPreference = options.readPreference ?? ReadPreference.primary;
-    this.selectServer(readPreferenceServerSelector(readPreference), options, (err, server) => {
-      if (err) {
-        this.close();
-
-        typeof callback === 'function' ? callback(err) : this.emit(Topology.ERROR, err);
-        return;
-      }
-
-      // TODO: NODE-2471
-      if (server && this.s.credentials) {
-        server.command(ns('admin.$cmd'), { ping: 1 }, err => {
+    // need to store the validated options for the closure or typescript complains
+    const closureOptions = options;
+    const readPreference = closureOptions.readPreference ?? ReadPreference.primary;
+    process.nextTick(() => {
+      this.selectServer(
+        readPreferenceServerSelector(readPreference),
+        closureOptions,
+        (err, server) => {
           if (err) {
+            this.close();
```

Comment:
I don't think this `close` should be here, did adding it fix a test that was failing or something?

Reply:
Huh thanks for spotting... that might just be a remnant of a rebase gone wrong. I'll kick it out.